tinyxml2.cpp

```diff
@@ -204,7 +204,7 @@ char* StrPair::ParseText( char* p, const char* endTag, int strFlags, int& curLin
             Set( start, p, strFlags );
             return p + length;
         } else if (*p == '\n') {
-            ++curLineNum;
+            ++(*curLineNumPtr);

```

Comment:
What's the use of this change? It was a "pass by reference" for an int and now you pass address of int by value.